applications/inference_monitoring_streamlit.py

```diff
@@ -0,0 +1,228 @@
+
+        this_session = None
+
+        current_server = Server.get_current()
+        if hasattr(current_server, '_session_infos'):
+            # Streamlit < 0.56
+            session_infos = Server.get_current()._session_infos.values()
+        else:
+            session_infos = Server.get_current()._session_info_by_id.values()
+
+        for session_info in session_infos:
+            s = session_info.session
+            if (
+                # Streamlit < 0.54.0
+                (hasattr(s, '_main_dg') and s._main_dg == ctx.main_dg)
+                or
+                # Streamlit >= 0.54.0
+                (not hasattr(s, '_main_dg') and s.enqueue == ctx.enqueue)
+                or
+                # Streamlit >= 0.65.2
+                (not hasattr(s, '_main_dg') and s._uploaded_file_mgr == ctx.uploaded_file_mgr)
+            ):
+                this_session = s
+
+        if this_session is None:
+            raise RuntimeError(
+                "Oh noes. Couldn't get your Streamlit Session object. "
+                'Are you doing something fancy with threads?')
+
+        # Got the session object! Now let's attach some state into it.
+
+        if not hasattr(this_session, '_custom_session_state'):
+            this_session._custom_session_state = SessionState(**kwargs)
+
+        return this_session._custom_session_state
+
+def unique(list1):
+     
+    # insert the list to the set
+    list_set = set(list1)
+    # convert the set to the list
+    unique_list = (list(list_set))
+    return(unique_list)
+     
+def run():
+    st.title("Inference inspector")
+
+    symbol_set = st.sidebar.selectbox('symbol_set', SYMBOL_LIST)
+    arpabet = st.sidebar.number_input('arpabet', 0.)
+    model_format = st.sidebar.selectbox("model_format", MODEL_FORMAT)
+    
+    @st.cache(allow_output_mutation=True)
+    def get_vocoder():
+        return []
+    
+    @st.cache(allow_output_mutation=True)
+    def get_model():
+        return []
+    
+    @st.cache(allow_output_mutation=True)
+    def get_vc():
+        return []
+    
+    vocoder_path = st.text_input("Add vocoder path", "/mnt/disks/uberduck-experiments-v0/uberduck-ml-dev/nbs/test/fixtures/models/gen_02640000_studio")
+    if st.button("Add vocoder"):
+        get_vocoder().append(vocoder_path)
+    vocoder_path = st.sidebar.selectbox('Vocoder', unique(get_vocoder()))
+    
+    model_path = st.text_input("Add model path","/mnt/disks/uberduck-experiments-v0/gcs_logbucket/tensorboard-ubstudio_v3_20epochcheckpoints/model/tacotron2_20.pt")
+    if st.button("Add model"):
+        get_model().append(model_path)
+    model_path = st.selectbox('Model', unique(get_model()))
+
+    vc_path = st.text_input("Add vocoder config path","/mnt/disks/uberduck-experiments-v0/uberduck-ml-dev/nbs/test/fixtures/models/config_v1.json")
+    if st.button("Add vocoder config"):
+        get_vc().append(vc_path)
+    vc_path = st.sidebar.selectbox('Vocoder config', unique(get_vc()))
```

Comment:
These files don't actually exist on uberduck-ml-dev so I'd suggest putting a placeholder like `/path/to/config_v1.json` or `/path/to/tacotron.pt`